### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by elenapetra**

### CODE REVIEW

File: src/components/CourseForm/CourseForm.tsx
Comment (dav-dio) [inline src/components/CourseForm/CourseForm.tsx:1]: Button should say "Update course" when user is in update mode

File: src/components/CourseForm/CourseForm.tsx [hunk 1]
```diff
@@ -30,21 +33,31 @@ export const CreateCourse = () => {
...
-  };
-
-  const navigate = useNavigate();
-  const dispatch = useDispatch();
 
   useEffect(() => {
-    setAuthorsList(authorList);
-  }, [authorList]);
+    setCourseAuthors((prevState) => {
+      const prevStateIds = prevState.map((a) => a.id);
```
Comment (dav-dio) [inline src/components/CourseForm/CourseForm.tsx]: Mapping the whole entire previous state just to get the IDs is a bit wasteful.
Try something like:

```
useEffect(() => {
  setCourseAuthors((prevState) => {
    return [
      ...prevState,
      ...storeAuthors.filter(
        (author) =>
          course.authors.includes(author.id) &&
          !prevState.some(item => item.id === author.id)
        ),
      ];
    })
}, [storeAuthors])
```

File: src/components/Courses/Components/CourseCard/CourseCard.tsx [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
   const onCourseClick = (courseId: String) => {
     navigate(`/courses/${courseId}`);
   };
-  const dispatch = useDispatch();
+
   const handleDeleteClick = () => {
-    dispatch(deleteCourseAction(course.id));
+    dispatch(deleteCourseThunk(course.id));
+  };
+  const handleEditClick = () => {
```
Comment (dav-dio) [inline src/components/Courses/Components/CourseCard/CourseCard.tsx:28]: nit: If the edit button leads somewhere, it's a good practice to make it into a (router) link so that the user can see where it's going when they hover over it. MUI has a "component" prop, so you can import Link from react-router-dom and just to <Button component={Link} to={`/courses/${courseId}`} />

File: src/components/Login/Login.tsx [hunk 3]
```diff
@@ -53,9 +51,19 @@ export const Login = () => {
       } else {
         setLoginError(error.message);
       }
+      return false;
     }
+    return true;
   }
 
+  const onLogin = async () => {
+    const isLogged = await login();
```
Comment (dav-dio) [inline src/components/Login/Login.tsx]:
```suggestion
    const isLoggedIn = await login();
```

File: src/index.tsx [hunk 4]
```diff
@@ -10,9 +10,9 @@ const root = ReactDOM.createRoot(
   document.getElementById('root') as HTMLElement
 );
 root.render(
-  <React.StrictMode>
-    <Provider store={store}>
-      <App />
-    </Provider>
-  </React.StrictMode>
+  // <React.StrictMode>
```
Comment (dav-dio) [inline src/index.tsx]: Why did you remove Strict Mode? This is not recommended for development

File: src/services.ts [hunk 5]
```diff
@@ -1,16 +1,29 @@
...
         'Content-Type': 'application/json',
+        Authorization: `${token}`,
       },
     });
     if (!response.ok) {
       throw new Error('Failed to fetch courses');
     }
     const result = await response.json();
-    return result;
+    return result.result;
```
Comment (dav-dio) [inline src/services.ts]: To avoid result.result, which looks odd, you can do:

```
const { result } = await response.json();
return result;
```

**Review by dav-dio — APPROVED** [review]

File: src/components/Login/Login.tsx (hunk 3 above)
Comment (elenapetra) [reply src/components/Login/Login.tsx]: I updated the variable name to `isLoggedIn`.Thanks!

File: src/services.ts (hunk 5 above)
Comment (elenapetra) [reply src/services.ts]: I've made the recommended changes.Thanks!

File: src/components/CourseForm/CourseForm.tsx
Comment (elenapetra) [reply src/components/CourseForm/CourseForm.tsx:1]: I have changed the button label based on user mode.

File: src/components/CourseForm/CourseForm.tsx (hunk 1 above)
Comment (elenapetra) [reply src/components/CourseForm/CourseForm.tsx]: I have changed the code based on your suggestion. Thanks!

File: src/components/Courses/Components/CourseCard/CourseCard.tsx (hunk 2 above)
Comment (elenapetra) [reply src/components/Courses/Components/CourseCard/CourseCard.tsx:28]: I've made the suggested changes to the edit button.

File: src/index.tsx (hunk 4 above)
Comment (elenapetra) [reply src/index.tsx]: I had a bag in reducers: when fetching and storing courses and authors I spread the elements of the current state  followed by the elements of the action.payload array , so the data was double rendered . I thought it was because of strict mode but actually I shouldn't return the elements of the current state but only the elements from action.payload array. Now i fixed the bag and added Strict Mode. Thanks!
